sklearn/feature_extraction/text.py

```diff
@@ -676,12 +678,8 @@ def __init__(self, input='content', encoding='utf-8',
         if max_df < 0 or min_df < 0:
             raise ValueError("negative value for max_df or min_df")
         self.max_features = max_features
-        if max_features is not None:
-            if (not isinstance(max_features, numbers.Integral) or
-                    max_features <= 0):
-                raise ValueError(
-                    "max_features=%r, neither a positive integer nor None"
-                    % max_features)
+        if max_features < 0:
```

Comment:
We still need to support the legacy `max_features=None` as users may have been doing this. That setting will break here in Py3k. Please add a test.